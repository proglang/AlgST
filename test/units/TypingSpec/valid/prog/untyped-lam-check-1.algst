double : Int -> Int
double = \i -> i * 2
