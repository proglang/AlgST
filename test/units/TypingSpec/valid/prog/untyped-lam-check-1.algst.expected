double#M.0
|
`- ValueDecl
   |
   +- type
   |  |
   |  `- Type.Arrow U
   |     |
   |     +- Type.Type
   |     |  |
   |     |  `- TypeRef
   |     |     |
   |     |     `- Int
   |     |
   |     `- Type.Type
   |        |
   |        `- TypeRef
   |           |
   |           `- Int
   |
   +- params
   |
   `- definition
      |
      `- Exp.Abs U i#M.1
         |
         +- Type.Type
         |  |
         |  `- TypeRef
         |     |
         |     `- Int
         |
         `- Exp.App
            |
            +- Exp.App
            |  |
            |  +- Exp.Var (*)#Builtin.21
            |  |
            |  `- Exp.Var i#M.1
            |
            `- Exp.Lit
               |
               `- Lit.Int 2

